writer.go

```diff
@@ -73,6 +74,12 @@ type WriterConfig struct {
 	// The default is to use a target batch size of 100 messages.
 	BatchSize int
 
+	// Limit the maximum size of a request in bytes before being sent to
+	// a partition.
+	//
+	// The default is to use a kafka default value of 1048576.
+	MaxBytes int
```

Comment:
`MaxBytes` is too generic. `MaxMessageBytes`?

Reply:
Or, `MaxRequestBytes` I guess? I think the comment could include a note about exactly which data this includes, e.g. value + key and after serialization?